### DISCUSSION TITLE
Orienting long reads

### DISCUSSION THREAD

**Discussion by marchoeppner** [bio-raum/FooDMe2#33] [General]
ONT reads (and presumably Pacbio CCS as well) come in random orientation, which creates some issues during ASV clustering (redundant clusters). While this is not per se a deal breaker, it seems like orienting reads consistently would be preferable.  

I can see two options for that:

1) Vsearch with `vsearch -orient`  see [here](https://github.com/torognes/vsearch/releases/download/v2.28.1/vsearch_manual.pdf)
2) DADA2 with`removePrimers` see [here](https://rdrr.io/github/benjjneb/dada2/man/removePrimers.html)

Neither of these options will nicely integrate into FooDMe2 "as is". 

Vsearch: requires a reference database (Fasta or udb). While we do have that already for MIDORI (i.e. Fasta files are downloaded during installation) creating such a database from Genbank would basically double the storage requirements to well over 1TB. I am guessing searching that database with vsearch will blow up the compute node. So that option wouldn't be "universal" and thus problematic. 

Dada2: uses the orientation of primers to determine the "correct" orientation of the read. However, this process would replace Cutadapt and also requires exactly one forward and one reverse primer sequence as command line option (as opposed to reading it from a Fasta file). This approach wouldn't work for strategies with multiple primers (like the amniote primer set). So again, not really a workable solution...(at least not with some added complexity)

Maybe you have another/better idea?

**Comment by gregdenay** [comment]
One could add a post-clustering step where cluster sequences are screened for RC pairs and merged? 
A simple python script could even do the trick.
